kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java

```diff
@@ -15,60 +15,101 @@
  */
 package io.fabric8.kubernetes.client;
 
+import java.util.Locale;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.api.model.Namespaced;
 import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.utils.ApiVersionUtil;
+import io.fabric8.kubernetes.client.utils.Pluralize;
+import io.fabric8.kubernetes.model.annotation.Plural;
+import io.fabric8.kubernetes.model.annotation.Singular;
 import io.sundr.builder.annotations.Buildable;
-import lombok.ToString;
+
+import static io.fabric8.kubernetes.client.utils.Utils.isNullOrEmpty;
 
 /**
  * A base class for implementing a custom resource kind
  */
 @JsonDeserialize(
-    using = JsonDeserializer.None.class
+  using = JsonDeserializer.None.class
 )
-@ToString
+@JsonPropertyOrder({
+  "apiVersion",
+  "kind",
+  "metadata",
+  "spec",
+  "status"
+})
 @Buildable(builderPackage = "io.fabric8.kubernetes.api.builder", editableEnabled = false)
-public abstract class CustomResource implements HasMetadata {
-  private String kind;
-  private String apiVersion;
+public abstract class CustomResource<Spec extends KubernetesResource, Status extends KubernetesResource> implements HasMetadata {
```

Comment:
Please use single letters for [generic types](https://docs.oracle.com/javase/tutorial/java/generics/types.html) I almost went crazy trying to figure out where the Status class was.

![image](https://user-images.githubusercontent.com/488391/102221895-035f2f00-3ee3-11eb-903b-103ca9a68378.png)

Reply:
Will do. I was a little skeptical of the Sonar rule but apparently it is indeed a problem. My IDE shows the parameters quite well…

Reply:
:sweat_smile: I literally had to checkout your branch to understand what was going on.

I was afraid that some sort of `Status` class was introduced in this PR which is what triggered my alarm in the first place. 

After searching through the PR files several times with two browsers and going nuts while yelling at the GitHub web interface I ended up checking in my IDE and was then when I realized the problem :sweat_smile: :sweat_smile: :sweat_smile:

Reply:
😰 
Sorry about that. Fixed now.